### PR TITLE
Add CFF Citation file

### PR DESCRIPTION
**Pull request by julian-christl**
### Checklist
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).

### Motivation and Context
Closes https://github.com/ls1intum/Artemis/issues/3892

### Description
Added a citation file that shows the citation that should get used when referencing Artemis in a paper. Github has a widget for this. See the screenshots below. Sadly Github doesn't support all of the CFF standard so I removed some stuff again as discussed with @jpbernius 

### Steps for Testing
1. Check the file itself for consistency with https://dl.acm.org/doi/10.1145/3159450.3159602
2. Check that the citation gets shown correctly in the code tab [when selecting this branch](https://github.com/ls1intum/Artemis/tree/feature/add-citation-file) (APA and BibTeX

### Screenshots
307ab47a6bbd5078d266dda2a7aaa04f
998108037fa40121a9d6b599cf91c10e

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    - "Instant Feedback"
+    - "Online Editor"
+    - "Interactive Exercise Instructions"
+    - "Online Courses"
+    - "In-class Exercises"
+  url: "https://doi.org/10.1145/3159450.3159602"
+  abstract: "The increasing number of students in computer science courses leads to high efforts
+             in manual assessment of exercises. Existing assessment systems are not designed for
+             exercises with immediate feedback in large classes. In this paper, we present an AuTomated
+             assEssment Management System for interactive learning.ArTEMiS assesses solutions to
```
Comment (MaisiKoleni) [inline CITATION.cff:60]:
```suggestion
             assEssment Management System for interactive learning. ArTEMiS assesses solutions to
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+             programming exercises automatically and provides instant feedback so that students
+             can iteratively solve the exercise. It is open source and highly scalable based on
+             version control, regression testing and continuous integration. ArTEMiS offers an
+             online code editor with interactive exercise instructions, is programming language
+             independent and applicable to a variety of computer science courses. By using it,
+             students gain experiences in version control, dependency management and continuous
+             integration.We used ArTEMiS in 3 university and 1 online courses and report about
```
Comment (MaisiKoleni) [inline CITATION.cff:67]:
```suggestion
             integration. We used ArTEMiS in 3 university and 1 online courses and report about
```

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  - artemis
+  - double-blind
+  - text-exercises
+  - modeling-exercises
+  - apollon
+  - exam-mode
+  - automatic-feedback
+license: MIT
+version: 5.1.3
+message: "If you want to reference ArTEMis in your work, please cite this paper."
```
Comment (MaisiKoleni) [inline CITATION.cff:27]:
```suggestion
message: "If you want to reference Artemis in your work, please cite this paper."
```

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+repository-code: "https://github.com/ls1intum/Artemis/"
+title: "ArTEMiS"
```
Comment (MaisiKoleni) [inline CITATION.cff:29]:
```suggestion
title: "Artemis"
```

File: CITATION.cff [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      family-names: Seitz
+      given-names: Andreas
+  title: "ArTEMiS: An Automatic Assessment Management System for Interactive Learning"
+  doi: 10.1145/3159450.3159602
+  year: 2018
+  isbn: "9781450351034"
+  publisher:
+    name: "Association for Computing Machinery"
+  keywords:
+    - "Automated Assessment"
```
Comment (MDK5) [inline CITATION.cff:47]: All of these are lowercase in the original, keeping it that way would make sense

**Review by MDK5 — APPROVED** [review]
Reviewed in testing session, only found one nitpick:

File: CITATION.cff (hunk 2 above)
Comment (ndricimrr) [inline CITATION.cff:67]:
```suggestion
             integration.We used ArTEMiS in 3 universities and 1 online courses and report about
```

**Review by MaisiKoleni — CHANGES_REQUESTED** [review]
Looks good, but there are some minor issues, see inline code suggestions.

**Review by MDK5 — COMMENTED** [review]
Probably a github issue, there is a noticable difference from the bibtex citation from the dio page and the one from github:
github:
>@inproceedings{Krusche_ArTEMiS_An_Automatic,
	author = {Krusche, Stephan and Seitz, Andreas},
	doi = {10.1145/3159450.3159602},
	pages = {284--289},
	title = {{ArTEMiS: An Automatic Assessment Management System for Interactive Learning}},
	url = {https://doi.org/10.1145/3159450.3159602}
}

dio:
>@inproceedings{krusche2018artemis,
	author = {Krusche, Stephan and Seitz, Andreas},
	title = {ArTEMiS: An Automatic Assessment Management System for Interactive Learning},
	year = {2018},
	isbn = {9781450351034},
	publisher = {Association for Computing Machinery},
	address = {New York, NY, USA},
	url = {https://doi.org/10.1145/3159450.3159602},
	doi = {10.1145/3159450.3159602},
	abstract = {The increasing number of students in computer science courses leads to high efforts in manual assessment of exercises. Existing assessment systems are not designed for exercises with immediate feedback in large classes. In this paper, we present an AuTomated assEssment Management System for interactive learning.ArTEMiS assesses solutions to programming exercises automatically and provides instant feedback so that students can iteratively solve the exercise. It is open source and highly scalable based on version control, regression testing and continuous integration. ArTEMiS offers an online code editor with interactive exercise instructions, is programming language independent and applicable to a variety of computer science courses. By using it, students gain experiences in version control, dependency management and continuous integration.We used ArTEMiS in 3 university and 1 online courses and report about our experiences. We figured out that ArTEMiS is suitable for beginners, helps students to realize their progress and to gradually improve their solutions. It reduces the effort of instructors and enhances the learning experience of students.},
	booktitle = {Proceedings of the 49th ACM Technical Symposium on Computer Science Education},
	pages = {284–289},
	numpages = {6},
	keywords = {instant feedback, programming exercises, online editor, automated assessment, interactive exercise instructions, continuous integration, version control, online courses, in-class exercises},
	location = {Baltimore, Maryland, USA},
	series = {SIGCSE '18}
}

File: CITATION.cff (hunk 2 above)
Comment (julian-christl) [reply CITATION.cff:67]: I would not change the abstract here and keep it as it is in the original.

**Comment by julian-christl** [comment]
@MDK5 I tried to do some improvements but yeah, seems like GitHub decided to only do one have of the implementation...
The "--" is CFF style though as well as {{...}} for the title.

For some reason, it just doesn't display e.g. the year even though it works with the GitHub example. Also, it displays the journal (which is wrong here) but not the collection title. Should I leave it in so it gets at least displayed or remove it since it's not accurate?

Also, cc @jpbernius since this doesn't seem to work as easily as expected, what do you suggest here?

**Review by MaisiKoleni — APPROVED** [review]
Looks good to me now.

**Review by jhainzlschmid — COMMENTED** [review]
The citations look good to me. I find the "View citation file" button a bit confusing though. When I click it it does nothing. Do you think it is possible to review the button?

image

File: CITATION.cff (hunk 5 above)
Comment (julian-christl) [reply CITATION.cff]: Removed them now anyway

File: CITATION.cff (hunk 4 above)
Comment (julian-christl) [reply CITATION.cff]: done

File: CITATION.cff (hunk 2 above)
Comment (julian-christl) [reply CITATION.cff]: Removed the abstract

**Comment by julian-christl** [comment]
I discussed the issues with @jpbernius and we decided to proceed with the minimal support Github provides here.  So while it might not be the same as on the doi page, it should be enough to cite.

I added therefore the `journal` key even though it should be `collection-title` but Github doesn't like that.

I know that the button to show the citation file doesn't work but that's on Github. If people are interested, they can find the file in the root anyway.

**Review by MDK5 — APPROVED** [review]
While not perfect (mainly publisher, place, and series are still missing), this looks good to me given githubs limitations.

**Review by jhainzlschmid — APPROVED** [review]
Looks good to me. Thanks for investing the effort 👍

**Pull request merged**